overlord/devicestate/devicestate.go

```diff
@@ -514,10 +516,48 @@ func remodelEssentialSnapTasks(ctx context.Context, st *state.State, ms modelSna
 	return addExistingSnapTasks(st, ms.newSnap)
 }
 
+// collect all prerequisites of a given snap from its task set
+func prereqsFromSnapTaskSet(ts *state.TaskSet) ([]string, error) {
+	for _, t := range ts.Tasks() {
+		// find the first task that carries snap setup
+		sup, err := snapstate.TaskSnapSetup(t)
+		if err != nil {
```

Comment:
(nitpick^3) this could be: 
```
if err != nil && err != state.ErrNoState {
    return nil, err
}
continue
```
(which is what we doing in most other places like this iirc). But feel free to ignore (and maybe I'm missing somethng):

Reply:
there's a continue at the end `if err != nil { }` scope, so the change would have to look like:
```
if err != nil && err != state.ErrNoState {
    return nil, err
} 
if err != nil {
    // try the next one
    continue
}
```
which is probably not worth it